### DISCUSSION TITLE
PNNL Material Compendium + Natural Abundance of Elements Expansion for Neutron XS Libraries

### DISCUSSION THREAD

**Discussion by SterlingButters** [idaholab/MontePy#531] [Ideas]
Thought I would just formalize the feature request before someone beats me to it. The abundance expansion shouldn't be too difficult although if the goal is to stay away from 3rd party libs both implementations will be a lot of grunt work.

**Comment by MicahGale** [comment]
Could you link to the report? I think this would require converting the report into a JSON format.

I am a bit on the fence about including that in MontePy directly. I see there are some tools like this: for [fusion at least](https://github.com/fusion-energy/neutronics_material_maker?tab=readme-ov-file). Also I thought there was a PNNL specific library, but I can't find it.

I am leaning towards maybe working with those projects to provide the material definition. 

Thoughts @shimwell?

I do think this should probably hold off until the Material API is in better shape. It's still in the state of my first pass at implementation, and I don't like the dictionary interface.

**Comment by MicahGale** [comment]
Also natural abundances do we want to use the 2013 IUPAC report? That's what OpenMC uses.

**Comment by shimwell** [comment]
A JSON file of the materials compendium (and other formats) were distributed a while back ([see comment](https://github.com/openmc-dev/openmc/pull/2105#issuecomment-1689571713)) but it looks like the link has changed

However there is also a copy of the JSON in the source code for this nice tool for materials

https://github.com/pyne/materials-compendium

**Comment by MicahGale** [comment]
Thanks @shimwell. Any thoughts in how to integrate this with `neutronics_material_maker`? 

I think ultimately we would just use the OpenMC data. But now that I know PyNE can do this too, I think we need to think about how to make our implementation distinct and how we "add value"

**Comment by MicahGale** [comment]
Ok I really like the PyNE materials compendium. I am a little conflicted on the best way to do this, especially given my "keep it vanilla" approach.

I see three possible approaches:
1. Add `materials_compendium` as an (optional) dependency to MontePy and handle the material definition generation natively
2. Add MontePy as a dependency of `materials_compendium` and have it spit out MontePy material objects.
3. Make a third-party package that imports both and then spits out MontePy material objects.

Do you have thoughts @gonuke from a PyNE perspective?

**Comment by MicahGale** [comment]
A point that @tjlaboss brought up recently: is that many naturally occurring isotopes are not included in ENDF/B (and what's in each release varies even more). So to do this properly you need to know what nuclear data library is being used, and what isotopes that library has. This would require reading the `xsdir` file most likely (#113), and creates a lot of difficulty when switching libraries that OpenMC has been dealing with. 

I think for now this be postponed. I would be in favor though of a lightweight wrapper script for MontePy that would include these data.

**Reply by SterlingButters** [reply]
By "naturally occurring isotopes" do you mean elemental compositions?

**Comment by MicahGale** [comment]
https://github.com/openmc-dev/openmc/pull/4134 this PR could be helpful long term.
